### PR TITLE
Updates and refactors the bigip pool module to use REST

### PR DESCRIPTION
**Pull request by caphrim007**
##### SUMMARY
This patch updates the bigip_pool module to use F5's REST API
as well as includes unit tests.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

lib/ansible/modules/network/f5/bigip_pool.py

##### ANSIBLE VERSION

```
ansible 2.4.0
  config file = 
  configured module search path = [u'/Users/trupp/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/trupp/src/envs/ansible-2.0/lib/python2.7/site-packages/ansible
  executable location = /Users/trupp/src/envs/ansible-2.0/bin/ansible
  python version = 2.7.10 (default, Feb  7 2017, 00:08:15) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```

##### ADDITIONAL INFORMATION
Unit tests are provided. Integration tests can be found here

https://github.com/F5Networks/f5-ansible/blob/devel/test/integration/bigip_pool.yaml#L23
https://github.com/F5Networks/f5-ansible/tree/devel/test/integration/targets/bigip_pool/tasks

### CODE REVIEW

**Comment by caphrim007** [comment]
@gundalow this is ready for review, but is showing as broken because this proposal here is not yet accepted. https://github.com/ansible/proposals/issues/65

File: lib/ansible/modules/network/f5/bigip_pool.py [hunk 1]
```diff
@@ -222,370 +177,564 @@
       state: "absent"
```
Comment (gundalow) [inline lib/ansible/modules/network/f5/bigip_pool.py:177]: `state:` appears to have been removed, though is still used in examples.

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/f5/bigip_pool.py (hunk 1 above)
Comment (caphrim007) [reply lib/ansible/modules/network/f5/bigip_pool.py:177]: @gundalow this is a common arg

https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/f5_utils.py#L187

Comment (gundalow) [reply lib/ansible/modules/network/f5/bigip_pool.py:177]: Ah, thanks

**Review by gundalow — APPROVED** [review]

**Pull request merged**